controllers/infrastructure/byomachine_controller_test.go

```diff
@@ -685,6 +779,73 @@ var _ = Describe("Controllers/ByomachineController", func() {
 				Expect(k8sClientUncached.Delete(ctx, byoHost2)).Should(Succeed())
 			})
 		})
+
+		Context("When installer config template exists", func() {
+			BeforeEach(func() {
+				k8sInstallerConfigTemplate = builder.K8sInstallerConfigTemplate(defaultNamespace, defaultK8sInstallerConfigTemplateName).
+					WithBundleRepo("projects.registry.vmware.com/cluster_api_provider_bringyourownhost").
+					WithBundleType("k8s").
+					Build()
+				Expect(k8sClientUncached.Create(ctx, k8sInstallerConfigTemplate)).Should(Succeed())
+				WaitForObjectsToBePopulatedInCache(k8sInstallerConfigTemplate)
+			})
+
+			It("should create installer config from the template", func() {
+				ph, err := patch.NewHelper(byoMachine, k8sClientUncached)
+				Expect(err).ShouldNot(HaveOccurred())
+				byoMachine.Spec.InstallerRef = &corev1.ObjectReference{
+					Kind:       "K8sInstallerConfigTemplate",
+					Namespace:  k8sInstallerConfigTemplate.Namespace,
+					Name:       k8sInstallerConfigTemplate.Name,
+					UID:        k8sInstallerConfigTemplate.UID,
+					APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				}
+				Expect(ph.Patch(ctx, byoMachine, patch.WithStatusObservedGeneration{})).Should(Succeed())
+
+				WaitForObjectToBeUpdatedInCache(byoMachine, func(object client.Object) bool {
+					return object.(*infrastructurev1beta1.ByoMachine).Spec.InstallerRef != nil
+				})
+
+				_, err = reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: byoMachineLookupKey})
+				Expect(err).Should(MatchError("no hosts found"))
+
+				createdK8sInstallerConfig := &infrastructurev1beta1.K8sInstallerConfig{}
+				err = k8sClientUncached.Get(ctx, byoMachineLookupKey, createdK8sInstallerConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(k8sInstallerConfigTemplate.Spec.Template.Spec).To(Equal(createdK8sInstallerConfig.Spec))
+				Expect(createdK8sInstallerConfig.GetAnnotations()[infrastructurev1beta1.K8sVersionAnnotation]).To(Equal(*machine.Spec.Version))
+			})
+
+			AfterEach(func() {
+				Expect(k8sClientUncached.Delete(ctx, k8sInstallerConfigTemplate)).Should(Succeed())
```

Comment:
Can't we add `GenerateName` in builder in place of creating the resource with the same name and then deleting it?

Reply:
The name for the installer config should be same as the name of byomachine as per contract so it will not work.

Reply:
But this is installer template. We are not using installer template name for creating the installer CR.

Reply:
Ohh then yes it can be done 👍